packages/relay-runtime/util/recycleNodesInto.js

```diff
@@ -30,29 +30,33 @@ function recycleNodesInto<T>(prevData: T, nextData: T): T {
   const prevArray = Array.isArray(prevData) ? prevData : null;
   const nextArray = Array.isArray(nextData) ? nextData : null;
   if (prevArray && nextArray) {
+    const isFrozen =
```

Comment:
@mike-marcacci we circled back on this while reviewing internally. You previously mentioned why you make the `isFrozen` check once per value, but it seems that given that we only do this in dev mode its fine if we just make an early check here:

```
+  // In dev builds, values may be frozen, in which case they cannot be recycled into.
+  if (__DEV__) {
+    if (Object.isFrozen(nextData) {
+      return nextData;
+    }
+  }
```

Reply:
Either way is fine by me! Am I correct in assuming that these dev flags get replaced with constants in the build process, and then all constant paths get compiled out? If so, then it really doesn't matter – you'll end up with no checks in production regardless of where they happen in dev.